### PR TITLE
Homework Tab for admin as well as user

### PR DESCRIPTION
**Pull request by abhishek2021005**

### CODE REVIEW

File: packages/alea-frontend/components/HomeworkManager.tsx [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+                sx={{
+                  borderRadius: '25px',
+                  marginLeft: '5px',
+                }}
+              >
+                {t.createHomework}{' '}
+              </Button>
+            </Box>
+
+            <TableContainer component={Paper} sx={{ maxHeight: '500px', overflowY: 'auto' }}>
```
Comment (abchugh) [inline packages/alea-frontend/components/HomeworkManager.tsx:202]: move this to a separate component. Preferably outside this file. As a rule of thumb, keep files smaller than 200 lines whenever possible.

File: packages/alea-frontend/components/HomeworkManager.tsx [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+                    );
+                  })}
+                </TableBody>
+              </Table>
+            </TableContainer>
+          </>
+        )}
+
+        {(view === 'create' || view === 'edit') && (
+          <Box
```
Comment (abchugh) [inline packages/alea-frontend/components/HomeworkManager.tsx:264]: separate component

File: packages/api/src/lib/homework-api.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+export async function createHomework(data: any) {
+  return axios.post('/api/homework/create-homework', data, {
+    headers: getAuthHeaders(),
+  });
+}
+
+export async function updateHomework(data: any) {
+  return axios.put('/api/homework/update-homework', data, {
```
[the remark this answers is not in the document]
Comment (abhishek2021005) [reply packages/api/src/lib/homework-api.ts:19]: Noted, fixed!

File: packages/alea-frontend/pages/api/homework/get-homework.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (req.method === 'GET') {
+    const courseId = req.query.courseId as string;
+    try {
+      const results = await executeDontEndSet500OnError(
+        'SELECT * FROM homework WHERE courseId = ? ORDER BY homeworkDate ASC',
+        [courseId],
+        res
+      );
+      console.log('resss', results);
```
[the remark this answers is not in the document]
Comment (abhishek2021005) [reply packages/alea-frontend/pages/api/homework/get-homework.ts:13]: Removed!

File: packages/alea-frontend/pages/api/homework/get-homework.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      if (results) {
```
[the remark this answers is not in the document]
Comment (abhishek2021005) [reply packages/alea-frontend/pages/api/homework/get-homework.ts:14]: understood and implemented!

File: packages/alea-frontend/components/HomeworkManager.tsx [hunk 6]
```diff
@@ -0,0 +1,354 @@
...
+  deleteHomework,
+} from '@stex-react/api';
+import Link from 'next/link';
+import { getLocaleObject } from '../lang/utils';
+import { useRouter } from 'next/router';
+
+const HomeworkManager = ({ courseId }) => {
+  const [homeworks, setHomeworks] = useState<HomeworkInfo[]>([]);
+  const [homeworkId, setHomeworkId] = useState<number | null>(null);
+  const [homeworkName, setHomeworkName] = useState<string>('');
```
[the remark this answers is not in the document]
Comment (abhishek2021005) [reply packages/alea-frontend/components/HomeworkManager.tsx:40]: Noted!

File: packages/alea-frontend/components/HomeworkManager.tsx [hunk 7]
```diff
@@ -0,0 +1,354 @@
...
+    };
+
+    try {
+      let response;
+      if (homeworkId) {
+        response = await updateHomework(body);
+      } else {
+        response = await createHomework(body);
+      }
+      console.log('resDD', response);
```
[the remark this answers is not in the document]
Comment (abhishek2021005) [reply packages/alea-frontend/components/HomeworkManager.tsx:86]: Done!

File: packages/alea-frontend/components/HomeworkManager.tsx [hunk 8]
```diff
@@ -0,0 +1,354 @@
...
+      setMessage(error.response?.data?.message || 'An error occurred');
+      setOpenSnackbar(true);
+    }
+  };
+
+  const handleEdit = (homework: HomeworkInfo) => {
+    setHomeworkId(homework.homeworkId);
+    setHomeworkName(homework.homeworkName);
+    const localDate = new Date(homework.homeworkDate);
+    const formattedDate = `${localDate.getFullYear()}-${String(localDate.getMonth() + 1).padStart(
```
[the remark this answers is not in the document]
Comment (abhishek2021005) [reply packages/alea-frontend/components/HomeworkManager.tsx:102]: Done!

File: packages/alea-frontend/components/HomeworkManager.tsx (hunk 1 above)
Comment (abhishek2021005) [reply packages/alea-frontend/components/HomeworkManager.tsx:202]: Refactored!

File: packages/alea-frontend/components/HomeworkManager.tsx (hunk 2 above)
Comment (abhishek2021005) [reply packages/alea-frontend/components/HomeworkManager.tsx:264]: Refactored!

**Pull request merged**